ddtrace/internal/telemetry/writer.py

```diff
@@ -0,0 +1,221 @@
+import logging
+from typing import ClassVar
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ddtrace.internal import forksafe
+
+from ..agent import get_connection
+from ..agent import get_trace_url
+from ..compat import get_connection_response
+from ..compat import httplib
+from ..encoding import JSONEncoderV2
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..utils.formats import get_env
+from ..utils.time import StopWatch
+from .request import create_telemetry_request
+from .request import get_headers
+
+
+log = get_logger(__name__)
+
+
+def _get_interval_or_default():
+    return float(get_env("instrumentation_telemetry", "interval", default=60))
+
+
+AGENT_URL = get_trace_url()
+ENDPOINT = "telemetry/proxy/api/v2/apmtelemetry"
+
+
+class TelemetryWriter(PeriodicService):
+    """
+    Periodic service which sends Telemetry request payloads to the agent-proxy
+    """
+
+    enabled = False  # type: ClassVar[bool]
+    _instance = None  # type: ClassVar[Optional[TelemetryWriter]]
+    _lock = forksafe.Lock()  # type: ClassVar[forksafe.ResetObject]
+    sequence = 0  # type: ClassVar[int]
+
+    def __init__(self):
+        # type: () -> None
+        super(TelemetryWriter, self).__init__(interval=_get_interval_or_default())
+
+        self._encoder = JSONEncoderV2()
+        self._events_queue = []  # type: List[Dict]
+        self._integrations_queue = []  # type: List[Dict]
+
+    def _send_request(self, request):
+        # type: (Dict) -> Optional[httplib.HTTPResponse]
+        """
+        Sends a telemetry request to the trace agent
+        """
+        conn = get_connection(AGENT_URL)
+        rb_json = self._encoder.encode(request)
+        resp = None
+        with StopWatch() as sw:
+            try:
+                conn.request("POST", ENDPOINT, rb_json, get_headers(request["request_type"]))
+                resp = get_connection_response(conn)
```

Comment:
We need to verify the response status code.

If there is a non-200/2 then we need to log a warning?